src/main/java/com/google/api/codegen/transformer/FeatureConfig.java

```diff
@@ -41,6 +47,24 @@
    */
   boolean useResourceNameConverters(FieldConfig fieldConfig);
 
+  /**
+   * Returns true if useResourceNameFormatOptionInSample() is true but
+   * resourceNameProtoAccessorsEnabled() is false.
+   */
+  boolean useResourceNameConvertersInSample(FieldConfig fieldConfig);
+
+  /**
+   * Returns true if useResourceNameFormatOptionInSampleOnly() is true but
+   * resourceNameProtoAccessorsEnabled() is false.
+   */
+  boolean useResourceNameConvertersInSampleOnly(FieldConfig fieldConfig);
+
+  /**
+   * Returns true if useResourceNameFormatOptionInSampleOnly() is true but
+   * resourceNameProtoAccessorsEnabled() is false.
+   */
+  boolean useResourceNameFormatOptionInSampleOnly(FieldConfig fieldConfig);
```

Comment:
What about a `useResourceNameFormatOptionInFlattenedSignature` method?

Reply:
That won't help the other branch in `InitCodeTransformer`, which needs to do a transformation when passing a resource name variable to a proto field.